### ISSUE TITLE
Class include other Class but with .nullish() or .optional() or .nullable()

### ISSUE THREAD

**Issue by hepr** [sam-goodwin/zod-class#28]
My classes

```
export class ProductClass extends Z.class({
entityId: z.string(),
productId: z.string(),
productMain: z.string().nullable().optional(),
maxStockLevel: z.number().nullable().optional(),
minStockLevel: z.number().nullable().optional(),
stockTimeLimit: z.number().nullable().optional(),
sellAccount: z.string().toUpperCase().trim().nullable().optional(),
weight: z.number().min(0).default(0),
....
}) {
}

```

```
export class ProductInventoryLocationClass extends Z.class({
warehouseId: z.string(),
location: z.string(),
...
product: ProductClass.nullish(), // Crash with .nullish()
}) {

}
```

`const productInventoryLocation = ProductInventoryLocationClass.parse(response)`

Crash only when product field with .nullish() or .optional() or .nullable()

v0.0.14

TypeError: Cannot read properties of undefined (reading '_parse')
❯ ZodOptional._parse node_modules/zod/lib/index.mjs:3765:36
❯ ZodNullable._parse node_modules/zod/lib/index.mjs:3784:36
❯ ZodObject._parse node_modules/zod/lib/index.mjs:2379:37
❯ ZodObject._parseSync node_modules/zod/lib/index.mjs:678:29
❯ ZodObject.safeParse node_modules/zod/lib/index.mjs:708:29
❯ ZodObject.parse node_modules/zod/lib/index.mjs:689:29
❯ Function.parse node_modules/zod-class/src/index.ts:306:38

**Comment by sam-goodwin** [comment]
Thanks for reporting the error, I am able to reproduce it. Will fix.

**Comment by sam-goodwin** [comment]
I've managed to fix the parse error but can't add `nullish` because of weird type errors.

I wouldn't recommend using this library because zod wasn't designed to be extended this way. Perhaps it'll be fixed in a later zod version.

**Comment by sam-goodwin** [comment]
Fixed in `0.0.15`. But, I would also caution against using this library.
